src/clojure/puppetlabs/ssl_utils/core.clj

```diff
@@ -587,6 +587,28 @@
   (= (.getIssuerX500Principal crl)
      (.getSubjectX500Principal cert)))
 
+(schema/defn ^:always-validate crl-issued-by-cert-in-chain? :- schema/Bool
+  "Given a CRL and a certificate chain, determine whether the CRL was issued by
+  any certificate in the chain."
+  [crl :- X509CRL
+   cert-chain :- [X509Certificate]]
+  (->> cert-chain
+       (some (partial crl-issued-by-cert? crl))
+       boolean))
+
+(schema/defn ^:always-validate verify-crls-against-cert-chain
+  "Given a CRL chain and a certificate chain, ensure all CRLs were issued by a
+  certificate in the chain, otherwise throw."
```

Comment:
Did we say we need to allow the case where people are submitting a CRL chain with other random stuff in it? If that's the case, maybe the logic we want is rather "For each of these certs, make sure there is a CRL in this chain."

Reply:
I think what we want to do is find the issuer chain of our cert and all their CRLs, ensuring those CRLs are valid and that none of the CAs in our chain of issuers has been revoked.